src/store/result/actions/result.actions.ts

```diff
@@ -0,0 +1,9 @@
+import { FETCH_PETS_REQUEST, FETCH_PETS_FAILTURE, FETCH_PETS_SUCCESS } from '../types/result.types';
+import { createAsyncAction } from 'typesafe-actions';
+import { Pet } from '../../../pages/result/result.interfaces';
+
+export const fetchPetsAsync = createAsyncAction(FETCH_PETS_REQUEST, FETCH_PETS_SUCCESS, FETCH_PETS_FAILTURE)<
```

Comment:
It is not necessary to declare type in a separate file. With typesafe-actions, you may use just strings for types

Reply:
I'll change it